src/AnalysisServices/AnalysisServices.Dataplane/Models/AsAzureDataplaneClient.cs

```diff
@@ -0,0 +1,172 @@
+        {
+            return await CallGetAsync(BaseUri, requestUrl, new Guid());
+        }
+
+        /// <summary>
+        /// Calls SendRequestAsync() for a POST.
+        /// </summary>
+        /// <param name="baseUri">The base Uri to call.</param>
+        /// <param name="requestUrl">The request Url.</param>
+        /// <param name="correlationId">The CorrelationId</param>
+        /// <param name="content">The content to post (optional).</param>
+        /// <returns>The http response message.</returns>
+        public async Task<HttpResponseMessage> CallPostAsync(Uri baseUri, string requestUrl, Guid correlationId, HttpContent content = null)
+        {
+            return await SendRequestAsync(HttpMethod.Post, baseUri, requestUrl, correlationId, content);
+        }
+
+        /// <summary>
+        /// Calls SendRequestAsync() for a POST using a blank correlationId.
+        /// </summary>
+        /// <param name="baseUri">The base Uri to call.</param>
+        /// <param name="requestUrl">The request Url.</param>
+        /// <param name="content">The content to post (optional).</param>
+        /// <returns>The http response message.</returns>
+        public async Task<HttpResponseMessage> CallPostAsync(Uri baseUri, string requestUrl, HttpContent content = null)
+        {
+            return await CallPostAsync(baseUri, requestUrl, new Guid(), content);
+        }
+
+        /// <summary>
+        /// Calls SendRequestAsync() for a POST using the default BaseUri and a blank correlationId.
+        /// </summary>
+        /// <param name="requestUrl">The Request Url.</param>
+        /// <param name="content">The content to post (optional).</param>
+        /// <returns>The http response message.</returns>
+        public async Task<HttpResponseMessage> CallPostAsync(string requestUrl, HttpContent content = null)
+        {
+            return await CallPostAsync(BaseUri, requestUrl, content);
+        }
+
+        #endregion
+
+        private async Task<HttpResponseMessage> SendRequestAsync(
+            HttpMethod method,
+            Uri baseUri,
+            string requestUrl,
+            Guid correlationId,
+            HttpContent content = null,
+            CancellationToken cancellationToken = default(CancellationToken))
+        {
+            // Create HTTP transport objects
+            HttpRequestMessage httpRequest = new HttpRequestMessage()
+            {
+                Method = method,
+                RequestUri = new Uri(baseUri, requestUrl),
+                Content = content
+            };
+
+            // Set Headers
+            AddHeader(httpRequest.Headers, "x-ms-client-request-id", correlationId.ToString());
+
+            // Set Credentials
+            if (Credentials != null)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+                await Credentials.ProcessHttpRequestAsync(httpRequest, cancellationToken).ConfigureAwait(false);
```

Comment:
can you explain to me how this works with respect to SendAsync? Thanks!

Reply:
This line creates and adds the authentication (bearer token) to the http request.